pkg/pool-hooks/contracts/MevHook.sol

```diff
@@ -0,0 +1,161 @@
+    }
+
+    /// @inheritdoc IHooks
+    function onRegister(
+        address,
+        address pool,
+        TokenConfig[] memory,
+        LiquidityManagement calldata
+    ) public override onlyVault returns (bool) {
+        _poolMevTaxMultipliers[pool] = _defaultMevTaxMultiplier;
+        _poolMevTaxThresholds[pool] = _defaultMevTaxThreshold;
+
+        return true;
+    }
+
+    /// @inheritdoc IHooks
+    function getHookFlags() public pure override returns (HookFlags memory) {
+        HookFlags memory hookFlags;
+        // The MEV Tax is charged as a swap fee. Searcher transactions pay a higher swap fee percentage.
+        hookFlags.shouldCallComputeDynamicSwapFee = true;
+        return hookFlags;
+    }
+
+    /// @inheritdoc IHooks
+    function onComputeDynamicSwapFeePercentage(
+        PoolSwapParams calldata,
+        address pool,
+        uint256 staticSwapFeePercentage
+    ) public view override returns (bool, uint256) {
+        if (_mevTaxEnabled == false) {
+            return (true, staticSwapFeePercentage);
+        }
+
+        // If gasprice is lower than basefee, the transaction is invalid and won't be processed. Gasprice is set
+        // by the transaction sender, is always bigger than basefee and the difference between gasprice and basefee
+        // defines the priority gas price (the part of the gas cost that will be paid to the validator).
+        uint256 priorityGasPrice = tx.gasprice - block.basefee;
+
+        // If priorityGasPrice < threshold, it means the transaction is from a retail user.
+        if (priorityGasPrice < _poolMevTaxThresholds[pool]) {
+            return (true, staticSwapFeePercentage);
+        }
+
+        uint256 mevSwapFeePercentage = priorityGasPrice.mulDown(_poolMevTaxMultipliers[pool]);
+
+        // If the resulting fee percentage is greater than MAX_FEE_PERCENTAGE, returns the max fee percentage.
+        if (mevSwapFeePercentage >= _MEV_MAX_FEE_PERCENTAGE) {
+            return (true, _MEV_MAX_FEE_PERCENTAGE);
+        }
+
+        // If static fee percentage is higher than mev fee percentage, uses the static one.
+        return (true, staticSwapFeePercentage > mevSwapFeePercentage ? staticSwapFeePercentage : mevSwapFeePercentage);
+    }
+
+    /// @inheritdoc IMevHook
+    function isMevTaxEnabled() external view returns (bool) {
+        return _mevTaxEnabled;
+    }
+
+    /// @inheritdoc IMevHook
+    function disableMevTax() external authenticate {
+        _mevTaxEnabled = false;
+    }
+
+    /// @inheritdoc IMevHook
+    function enableMevTax() external authenticate {
+        _mevTaxEnabled = true;
+    }
+
+    /// @inheritdoc IMevHook
+    function getDefaultMevTaxMultiplier() external view returns (uint256) {
+        return _defaultMevTaxMultiplier;
+    }
+
+    /// @inheritdoc IMevHook
+    function setDefaultMevTaxMultiplier(uint256 newDefaultMevTaxMultiplier) external authenticate {
+        _defaultMevTaxMultiplier = newDefaultMevTaxMultiplier;
+    }
```

Comment:
```suggestion
    function setDefaultMevTaxMultiplier(uint256 newDefaultMevTaxMultiplier) external authenticate {
        _setDefaultMevTaxMultiplier(newDefaultMevTaxMultiplier);
    }
    
    function _setDefaultMevTaxMultiplier(uint256 newDefaultMevTaxMultiplier) internal {
        if (newDefaultMevTaxMultiplier > MAX_MEV_TAX_MULTIPLIER) {
            revert MevTaxMultiplierTooHigh();
        }
        
        _defaultMevTaxMultiplier = newDefaultMevTaxMultiplier;
        
        emit DefaultMevTaxMultiplierSet(newDefaultMevTaxMultiplier);
    }
```
These should have validation and events. Here and for all the other setters.

Reply:
It has events, but no validations.
The reason is that the Multiplier and Threshold values are open ended right now and require research. We don't know the max value for these parameters, and both of them may be set to 0 for some reason.

The only thing that I could validate is if the MEV swap fee percentage is greater than max fee percentage.